### ISSUE TITLE
possible bug with HERReplayBuffer under pyrtorch

### ISSUE THREAD

**Issue by st2yang** [rlworkgroup/garage#1740]
Hi,

I managed to run examples/tf/her_ddpg_fetchreach.py with tuned parameters (PR #1739 ). Then I try using pytorch, and something strange happens. It was working fine at first then crashes dramatically. The comparison between [tf ](https://ibb.co/80FRg5y)and [torch](https://ibb.co/9yx8bx2).

The main modification is tf.policies.ContinuousMLPPolicy, tf.q_functions.ContinuousMLPQFunction -> torch.policies.DeterministicMLPPolicy, torch.q_functions.ContinuousMLPQFunction, while other parameters are kept the same.

**Comment by ryanjulian** [comment]
Great work!

Wow, this is a tough one. Do you know if this behavior persists across several random seeds?

DDPG can be a very unstable algorithm, and it's possible you're seeing some seed-dependent instability.

By the way, it's easy to share experiments from garage using tensorboard.dev. See [the docs](https://garage.readthedocs.io/en/latest/user/monitor_experiments_with_tensorboard.html#share-your-tensorboard-logs) for full details.

Other ideas:
* Turning down the learning rate
* Perhaps the replay buffer becomes over-populated with positive (high-reward) examples and doesn't have negative (low-reward) examples? You could check this by logging the mean/median/min/max (or a histogram) of the rewards in the batches the algorithm samples from the replay buffer. [The docs](https://garage.readthedocs.io/en/latest/user/monitor_experiments_with_tensorboard.html#log-scalar-values-to-tensorboard) cover how to do that.
* Collecting negatives even when the agent is succeeding is one purpose for the `exploration_policy`. Does it improve if you increase `sigma` in the exploration policy, or use a different exploration policy?

**Comment by ryanjulian** [comment]
@krzentner @maliesa96

**Comment by st2yang** [comment]
@ryanjulian This behavior persists under the default seed. But I didn't test it with different seeds. I can try to test possible cases when I have more time.

**Comment by irisliucy** [comment]
@st2yang Would you share your experiment settings/ essential code snippets so that we can reproduce the issues? Thanks!

**Comment by st2yang** [comment]
@irisliucy Sorry for the late reply. Can you try adapting examples/tf/her_ddpg_fetchreach.py to run with pytorch? I just changed some tf-related to be pytorch things, and training got worse. I used very similar parameters when debugging.

If the bug persists, you should see something similar.
